keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -73,608 +74,482 @@
+
+### Control Flows
+
+This section outlines the scenarios that COSI personas will initiate, and for what purpose. Each
+scenario includes enough detail to express the important interaction requirements between personas
+and the COSI system, and between COSI components. This section avoids unnecessarily naming specific
+API elements so as not to confuse complex system interaction requirements with specific
+implementation/spec details.
+
+#### Installing the COSI System
+
+Admin installs the COSI system and driver(s) to allow User self-service.
+
+1. Assume that a Vendor has already created a COSI driver
+2. Admin deploys the COSI controller
+3. Admin deploys vendor COSI driver
+4. Admin creates BucketClass and AccessClass configuring COSI and vendor driver features
+
+#### Creating a Bucket
+
+User self-provisions a bucket to store their workload's data.
+
+1. Admin allows User to use BucketClass
+2. User creates BucketClaim that uses BucketClass
+3. COSI controller observes BucketClaim
+   1. Controller applies a finalizer to the BucketClaim to prevent object deletion before cleanup
+   2. Controller then creates a new intermediate Bucket resource
+   3. Controller applies a finalizer to intermediate Bucket to prevent it from being deleted before
+      the BucketClaim is deleted
+   4. Controller copies BucketClaim and BucketClass parameters to Bucket
+   5. Bucket status `BucketReady` is false initially, indicating bucket is not yet provisioned in OSP
+   6. Controller is now waiting for the intermediate Bucket to be reconciled by COSI sidecar before
+      continuing with BucketClaim reconciliation
+4. COSI sidecar detects the intermediate Bucket resource
+   1. If the Bucket's driver matches the sidecar's driver, continue
+   2. Sidecar calls the COSI driver via gRPC to provision the OSP bucket
+   3. To ensure idempotency, gRPC call must include enough information to identify repeat requests
+      for the same Bucket
+      <!-- TODO: discuss this; namespace and name are not strictly enough info b/c a user could access
+      the same OSP from multiple k8s clusters. COSI system will need some kind of optional specified ID? -->
+5. If OSP returns provision fail, COSI sidecar reports error to Bucket status and retries gRPC call
+6. When OSP returns provision success, COSI sidecar updates Bucket status `BucketReady` to true
+7. Controller detects that the Bucket is provisioned successfully (`BucketReady`==true)
+   1. Controller finishes BucketClaim reconciliation processing/validation as needed
+   2. Controller applies `BucketReady` true status to BucketClaim, finishing BucketClaim reconcile
+
+The mechanism by which the Controller waits for the intermediate Bucket to be provisioned in the
+middle of BucketClaim reconciliation is not specified here. The important behavior (waiting) is
+defined, and the logic coordinating the init-wait-finish reconcile process is left as an
+implementation detail.
```

Comment:
It's probably better to define high-level behavior expectations here, rather than pigeonhole ourselves into a specific implementation solution.